controllers/idler/idler_controller.go

```diff
@@ -299,6 +301,24 @@ func (r *Reconciler) scaleDeploymentToZero(logger logr.Logger, namespace string,
 					return true, nil
 				}
 
+				return false, err
+			} else if errors.IsInternalError(err) && strings.Contains(err.Error(), "the spec replicas field \".spec.replicas\" does not exist") {
```

Comment:
What if they error message is changed in the future? Not sure we should rely on the message :( There is no any e2e test which would catch it. So just checking the error code is probably safer...

Reply:
yes true, the error message is subject to change so I have removed the check